src/transformers/models/informer/modeling_informer.py

```diff
@@ -806,21 +806,24 @@ def forward(
 
         # __prob_QK
         # calculate the sampled Q_K
-        K_expand = key_states.unsqueeze(2).expand(-1, L_Q, L_K, -1) # torch.Size([52, 14, 14, 4])
-        index_sample = torch.randint(0, L_K, (L_Q, U_part)) # torch.Size([14, 14])
+        # K_expand = key_states.unsqueeze(2).expand(-1, L_Q, L_K, -1) # torch.Size([52, 14, 14, 4])
+        index_sample = torch.randint(0, L_K, (U_part,)) # torch.Size([14])
```

Comment:
this is smart! In my implementation I fixed the sizes of `K_expand`, but in your implementation of `index_sample` we really don't need it. awesome!